tasks/main.yml

```diff
@@ -3,11 +3,11 @@
   delegate_to: localhost
 
 - include: add_host.yml
-  when: mr_provisioner_machine_name != False and mr_provisioner_do_provision
+  when: mr_provisioner_machine_name != inventory_hostname and mr_provisioner_do_provision
```

Comment:
We can safely assume that if the machine_name is the same as the inventory_hostname, not going through the ip getting thing should work well.
If we can't, then a new boolean needs to be introduced, as Ansible doesn't seem to be able to keep track of if a variable has been overridden from defaults.

Reply:
Thanks for the fix. Using inventory_hostname here looks good to me. Here is the only missing bit. Once it is fixed, we are good to merge.
 
```
diff --git a/tasks/main.yml b/tasks/main.yml
index d7cd49a..b5ff621 100644
--- a/tasks/main.yml
+++ b/tasks/main.yml
@@ -10,4 +10,4 @@
   wait_for_connection:
     delay: 60
     timeout: 3600
-  when: mr_provisioner_do_provision and mr_provisioner_machine_name != inventory_hostname
+  when: mr_provisioner_do_provision and mr_provisioner_machine_name == inventory_hostname
```